### PR TITLE
hey-mail: init at 1.1.0

### PR DESCRIPTION
**Pull request by wandersoncferreira**
Missing package useful to hey-mail subscribers

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117878` [1](https://github.com/Mic92/nixpkgs-review)

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117878` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hey-mail</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:1039]: Please put this into a separate commit with the message ``maintainers: add bartuka``.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:49]:
```suggestion
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:56]:
```suggestion
  version = "1.1.0";
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:69]: Does the binary break if it is stripped? If not please remove it.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:75]:
```suggestion
    runHook preUnpack
    unsquashfs "$src"
    cd squashfs-root
    runHook postUnpack
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:111]: Please intent the string with two spaces like I did above.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:108]:
```suggestion
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:104]:
```suggestion
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:81]:
```suggestion
  mkdir -p $out/lib $out/share/applications/ $out/share/icons/
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:84]: Don't we have that file doubled in the output? ln -s is probably preferred here.

File: maintainers/maintainer-list.nix
Comment (wandersoncferreira) [reply maintainers/maintainer-list.nix:1039]: Thanks for feedbacks @SuperSandro2000 , I have this other PR opened https://github.com/NixOS/nixpkgs/pull/117908 where I add me as maintainer, would be better to fix this too and solve the conflict later or remove this code and merge this after #117908 ?

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (wandersoncferreira) [reply pkgs/applications/networking/mailreaders/hey-mail/default.nix:67]: No harm caused. Removing from latest.

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117878` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hey-mail</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
@wandersoncferreira please rebase the PR after whatsapp-for-linux was merged with you as maintainer.

**Comment by wandersoncferreira** [comment]
> @wandersoncferreira please rebase the PR after whatsapp-for-linux was merged with you as maintainer.

@doronbehar Rebased.

**Comment by wandersoncferreira** [comment]
Result of `nixpkgs-review pr 117878` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hey-mail</li>
  </ul>
</details>

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:56]: It would be nice to have a comment here explaining where to scratch the download link, or better, add to the package an `updateScript` that would do it automatically, see an example for `zoom-us`.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (doronbehar) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:54]: Maybe the `rev` should be defined here, if it's sort of related to the `version`.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (wandersoncferreira) [reply pkgs/applications/networking/mailreaders/hey-mail/default.nix:56]: @doronbehar  The idea here is that the `updateScript` will pull info from the download link like `version` and `ref` dynamically? So if we trigger update it will try to build the package with the latest version available in the download link?

Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hey-mail/default.nix:56]: `zoom-us` has:

https://github.com/NixOS/nixpkgs/blob/560cc2af818e6f08c5ef316d2f1dedfa47aee51a/pkgs/applications/networking/instant-messengers/zoom-us/default.nix#L119

And this enables anyone to update it with:

```
nix-shell maintainers/scripts/update.nix --argstr package zoom-us --argstr commit true
```

Which with a certain mediator calls the update script which has:

https://github.com/NixOS/nixpkgs/blob/560cc2af818e6f08c5ef316d2f1dedfa47aee51a/pkgs/applications/networking/instant-messengers/zoom-us/update.sh#L1-L10

-----

Overall, this is a sophisticated way of showing anyone interested not only where the version of the package appears in upstream's website, it also automates the process of retrieving the version string and updating the nix expression of the package. It doesn't build the package you updated but with `--argstr commit true` it runs a `git commit` with a proper `--message` suitable for the change in version done, if any. In your case it might be a bit more complicated because as I see it, you have to update 3-4 strings in the expression:

1. The version.
2. The `rev` maybe?
3. The hash that's part of the URL.
4. The `sha256` of the downloaded file.

`update-source-version` usually updates easily only `version` and `sha256` of the downloaded link. In your case it might get complicated. If you think it's above your Nix / shell skills level, at least please make the effort of explaining in a comment what is `rev` and where do you find the download link (I didn't find it easily) and what is the source of the small hash in the URL, if it's likely to change in the next version of the package.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:85]:
```suggestion
    patchelf \
      --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
      --set-rpath $out $out/hey-mail
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:93]:
```suggestion
    install -Dm755 $out/hey-mail $out/bin/hey-mail
    
    wrapProgram $out/bin/hey-mail \
      --prefix LD_LIBRARY_PATH : "$librarypath"
```

this should be sufficient.

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/hey-mail/default.nix:102]:
```suggestion
    sed -i "s:^Icon=.*:Icon=hey-mail:" "$out/meta/gui/hey-mail.desktop"

    # desktop file
    cp "$out/meta/gui/hey-mail.desktop" "$out/share/applications/"
    ln -s "$out/meta/gui/icon.png" "$out/share/icons/hey-mail.png"
```

File: pkgs/applications/networking/mailreaders/hey-mail/default.nix
Comment (SuperSandro2000) [reply pkgs/applications/networking/mailreaders/hey-mail/default.nix:54]: Is it derived from the version?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
